apps/src/lib/levelbuilder/NewCourseFields.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {useState} from 'react';
+import HelpTip from '@cdo/apps/lib/ui/HelpTip';
+import PropTypes from 'prop-types';
+
+export default function NewCourseFields(props) {
+  const [selectedFamilyName, setSelectedFamilyName] = useState('');
+  const [newFamilyName, setNewFamilyName] = useState('');
+  const [versionedCourse, setVersionedCourse] = useState('');
+
+  return (
+    <div>
+      <label>
+        What family is this course a part of?
+        <select
+          value={selectedFamilyName}
+          style={styles.dropdown}
+          className="familyNameSelector"
+          onChange={e => {
+            setSelectedFamilyName(e.target.value);
+            props.setFamilyName(e.target.value);
+          }}
+          disabled={newFamilyName !== ''}
+        >
+          <option value={''}>(None)</option>
+          {props.families.map(familyOption => (
+            <option key={familyOption} value={familyOption}>
+              {familyOption}
+            </option>
+          ))}
+        </select>
+        <span>
+          or{' '}
+          <input
+            type="text"
+            value={newFamilyName}
+            className="familyNameInput"
+            style={styles.smallInput}
+            onChange={e => {
+              setNewFamilyName(e.target.value);
```

Comment:
should/could we disallow family names which are already taken when the user types one in here? I would assume they are expecting to get a new family name when they type in their own answer.

Reply:
also, should / could we reject entries that end in `-DDDD`?

Reply:
I like this idea. I'm going to track it and follow up with another PR. https://codedotorg.atlassian.net/browse/PLAT-1514